optuna/integration/allennlp.py

```diff
@@ -163,16 +360,44 @@ class AllenNLPPruningCallback(EpochCallback):
         monitor:
             An evaluation metric for pruning, e.g. ``validation_loss`` or
             ``validation_accuracy``.
+
     """
 
-    def __init__(self, trial: optuna.trial.Trial, monitor: str):
+    def __init__(
+        self,
+        trial: Optional[optuna.trial.Trial] = None,
+        monitor: Optional[str] = None,
+    ):
         _imports.check()
 
         if allennlp.__version__ < "1.0.0":
             raise Exception("AllenNLPPruningCallback requires `allennlp`>=1.0.0.")
 
-        self._trial = trial
-        self._monitor = monitor
+        if trial is not None and monitor is not None:
+            self._trial = trial
+            self._monitor = monitor
+        else:
+            _environment_variables = _get_environment_variables_for_trial()
+            study_name = _environment_variables["study_name"]
+            trial_id = _environment_variables["trial_id"]
+            monitor = _environment_variables["monitor"]
+            storage = _environment_variables["storage"] or None
```

Comment:
A comment for `or None` would be helpful.

Reply:
I revised the PR to check if `storage` is `None` 8d227cd with a specific message.
What do you think about this change? Do you think it'd be better to add more comments?

Reply:
Looks good, thanks!